File: portfolio-backend/src/main/java/fabrizio/lombardi/portfolio_backend/models/Skill.java
package fabrizio.lombardi.portfolio_backend.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import fabrizio.lombardi.portfolio_backend.models.enums.SkillField;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Skill {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String name;
    @Enumerated(EnumType.STRING)
    private SkillField skillField;
}

File: frontend/src/interfaces/Skill.ts
import type { SKILL_FIELD } from "./enums/SkillField";

export interface Skill {
  id: number;
  name: string;
  skillField: SKILL_FIELD;
}

export const SkillPayload: Skill = {
  id: 0,
  name: "",
  skillField: "FRONTEND",
};
